avocado/plugins/beaker_result.py

```diff
@@ -0,0 +1,137 @@
+"""
+beaker result plugin
+
+Sends results and logs to beaker using the harness API.
+https://beaker-project.org/docs/alternative-harnesses/index.html
+
+Finds the beaker API entry point using the BEAKER_LAB_CONTROLLER_URL
+environment variable.  Does nothing in case the variable is not set.
+
+"""
+import glob
+import os
+import pprint
+import urllib.error
+import urllib.parse
+import urllib.request
+
+from avocado.core.output import LOG_UI
+from avocado.core.plugin_interfaces import ResultEvents
+
+
+class BeakerResult(ResultEvents):
+
+    """
+    send test results to beaker test harness api
+    """
+
+    name = 'beaker'
+    description = 'report results to beaker'
+
+    beaker_url = None
+    job_id = None
+
+    def __init__(self, config):
+        self.config = config  # make lint happy
+        baseurl = os.environ.get('BEAKER_LAB_CONTROLLER_URL')
+        recipeid = os.environ.get('RSTRNT_RECIPEID')
+        taskid = os.environ.get('RSTRNT_TASKID')
+        if baseurl is None or recipeid is None or taskid is None:
+            return
+        baseurl = baseurl.rstrip('/')
+        self.beaker_url = baseurl + '/recipes/' + recipeid + '/tasks/' + taskid
+        LOG_UI.info("beaker: using API at %s (R:%s T:%s)", baseurl, recipeid, taskid)
+
+    def send_request(self, req):
+        LOG_UI.info('beaker: %s %s ...', req.method, req.full_url)
+        try:
+            res = urllib.request.urlopen(req)  # nosec
+            return res
+        except urllib.error.URLError as err:
+            LOG_UI.info('beaker: Failed: %s', err)
+            return None
+        except Exception as err:
+            # should not happen
+            LOG_UI.info('beaker: Oops: %s', err)
+            return None
+
+    def post_result(self, state):
+        reqdict = {
+            'path': "%s/%s" % (self.job_id, state.get('name')),
+        }
+
+        result = state.get('status').lower().capitalize()
+        if result == 'Cancel':
+            result = 'Skip'
+        if result not in ['Pass', 'Fail', 'Skip']:
```

Comment:
The canonical source of "OK" or "NOT OK" for Avocado tests is defined here:

https://github.com/avocado-framework/avocado/blob/99109fa10c527ff8cb34f9be7fcdb71f33371277/avocado/core/teststatus.py#L17

But I realize its usage is not straightforward, so I've proposed [this](https://github.com/avocado-framework/avocado/pull/5151).  Hopefully it will be accepted/merged soon and so that you can use it.

Reply:
Looking at this again after your change got merged.  Noticed it doesn't help much :(
This is a /beaker/ status list.  Beaker has no 'cancel', so this gets mapped to 'skip' first.
Then we check if the result is a valid beaker status, and anything unknown is mapped to 'fail'.